lib/manageiq/providers/workflows/builtin_methods.rb

```diff
@@ -14,6 +14,39 @@ def self.email(params = {}, _secrets = {}, _context = {})
           miq_task_status!(runner_context)
         end
 
+        def self.embedded_ansible(params = {}, _secrets = {}, _context = {})
+          repository_url, repository_branch, playbook_name, playbook_id = params.values_at("RepositoryUrl", "RepositoryBranch", "PlaybookName", "PlaybookId")
+
+          vars = params
+                 .slice("Hosts", "ExtraVars", "BecomeEnabled", "Timeout", "Verbosity", "CredentialId", "CloudCredentialId", "NetworkCredentialId", "VaultCredentialId")
+                 .transform_keys { |k| k.underscore.to_sym }
+
+          vars[:execution_ttl] = vars.delete(:timeout) if vars.key?(:timeout)
+          %i[credential_id cloud_credential_id network_credential_id vault_credential_id].each do |key|
+            new_key = key.to_s.gsub(/_id$/, '').to_sym
+            vars[new_key] = vars.delete(key) if vars.key?(key)
+          end
+
+          if playbook_id
+            playbook = ::ConfigurationScriptPayload.find_by(:id => playbook_id)
+            return BuiltinRunnner.error!({}, :cause => "Unable to find Playbook: Id: [#{playbook_id}] Repository: [#{repository.name}]") if playbook.nil?
+          else
+            repository = ::ConfigurationScriptSource.find_by(:scm_url => repository_url, :scm_branch => repository_branch)
+            return BuiltinRunnner.error!({}, :cause => "Unable to find Repository: URL: [#{repository_url}] Branch: [#{repository_branch}]") if repository.nil?
+
+            playbook = ::ConfigurationScriptPayload.find_by(:configuration_script_source => repository, :name => playbook_name)
+            return BuiltinRunnner.error!({}, :cause => "Unable to find Playbook: Name: [#{playbook_name}] Repository: [#{repository.name}]") if playbook.nil?
```

Comment:
These are going to find any ConfigurationScriptSource.  Should we limit this to embedded ansible specific subclasses?